webserver/src/main/java/io/kestra/webserver/services/ai/gemini/GeminiAiService.java

```diff
@@ -0,0 +1,362 @@
+        final TenantService tenantService,
+        final FlowService flowService
+    ) {
+        SequentialAgent kestraFlowBuilder = initializeAgents(geminiConfiguration, pluginRegistry, jsonSchemaGenerator, tenantService, flowService);
+
+        IN_MEMORY_RUNNER = new InMemoryRunner(kestraFlowBuilder);
+    }
+
+    private static LlmAgent createMostRelevantTypesAgent(PluginRegistry pluginRegistry, JsonSchemaGenerator jsonSchemaGenerator, Gemini model) {
+        List<String> excludedPluginTypes = List.of("storages", "secrets", "apps", "app-blocks", "charts", "data-filters", "data-filters-kpi");
+        Map<String, String> descriptionByType = pluginRegistry.plugins().stream()
+            .flatMap(plugin -> plugin.allClassGrouped().entrySet().stream().filter(e -> !excludedPluginTypes.contains(e.getKey())).map(Map.Entry::getValue).flatMap(Collection::stream))
+            .map(clazz -> Map.entry(clazz.getName(), Optional.ofNullable(((Class<?>) clazz).getDeclaredAnnotation(io.swagger.v3.oas.annotations.media.Schema.class))))
+            .filter(e -> !e.getValue().map(io.swagger.v3.oas.annotations.media.Schema::deprecated).orElse(false))
+            .collect(
+                Collectors.toMap(
+                    Map.Entry::getKey,
+                    e -> e.getValue()
+                        .map(io.swagger.v3.oas.annotations.media.Schema::title)
+                        .orElse("")
+                )
+            );
+        String pluginRelevanceInstructions;
+        String serializedPlugins;
+        try {
+            serializedPlugins = JacksonMapper.ofJson().writeValueAsString(List.of(descriptionByType.entrySet().stream().map(e ->
+                Map.of("type", e.getKey(), "description", e.getValue())
+            )));
+        } catch (JsonProcessingException e) {
+            log.error("Failed to serialize plugin types for Gemini AI agent", e);
+            serializedPlugins = "[]";
+        }
+
+        pluginRelevanceInstructions = """
+            You are part of a Kestra Flow builder service. You are the plugin specialist. Your goal is to identify the most relevant plugin types for building a Kestra Flow based on the user's intent from the below list considering both the technology and the description of the plugin type. Here are the rules:
+            - Be lenient in matching - consider matching by technology name e.g. mongo -> MongoDB, postgres -> PostgreSQL, ...
+            - If a loop through index is needed (for example to do something X times), use ForEach task. If looping until a condition is met, use LoopUntil task instead.
+            - Detect data fetching requirements. If it's needed, make sure to include some tasks to do so (for example, if required to fetch data from a database, use the appropriate database task, if an http call is needed, use a HTTP task). If the initial Flow doesn't provide data that the user wants to use or display, make sure to include a task that fetches it (HTTP, database, ...).
```

Comment:
```suggestion
            - If data fetching is required, include appropriate tasks (e.g., database or HTTP tasks). If the flow needs to obtain data not already available, include a relevant data-fetching task.
```
this is too long and largely redundant -- you can try to cut it and see if you get a worse or a better result. I think data fetching requirements are implied in **"Your task is to identify the most relevant plugin types for building a Kestra Flow based on the user's intent"**

Reply:
Your phrasing seems to shorten so I'll try to go for it. I think it's important to include something similar still as I was getting some results where it was just logging some samples (eg. display the weather of tomorrow was just adding a Log task saying "the weather is ...", I believe the AI "thought" it had the data available from training set to directly put the data in :)